packages/core/ios/Sources/RNDdSdkConfiguration.swift

```diff
@@ -176,3 +176,159 @@ extension NSArray {
+            let proxyConfig = configuration["proxy"] as? NSDictionary
+            let serviceName = configuration["serviceName"] as? NSString
+            let firstPartyHosts = configuration["firstPartyHosts"] as? NSArray
+
+            return DdSdkConfiguration(
+                clientToken: clientToken ?? String(),
+                env: env ?? String(),
+                applicationId: applicationId ?? String(),
+                nativeCrashReportEnabled: nativeCrashReportEnabled ?? DefaultConfiguration.nativeCrashReportEnabled,
+                nativeLongTaskThresholdMs: nativeLongTaskThresholdMs ?? DefaultConfiguration.nativeLongTaskThresholdMs,
+                longTaskThresholdMs: (longTaskThresholdMs != nil) ? longTaskThresholdMs! : DefaultConfiguration.longTaskThresholdMs,
+                sampleRate: sampleRate ?? DefaultConfiguration.sessionSamplingRate,
+                site: site.asSite(),
+                trackingConsent: trackingConsent.asTrackingConsent(),
+                telemetrySampleRate: telemetrySampleRate ?? DefaultConfiguration.telemetrySampleRate,
+                vitalsUpdateFrequency: vitalsUpdateFrequency.asVitalsUpdateFrequency(),
+                trackFrustrations: trackFrustrations ?? DefaultConfiguration.trackFrustrations,
+                uploadFrequency: uploadFrequency.asUploadFrequency(),
+                batchSize: batchSize.asBatchSize(),
+                trackBackgroundEvents: trackBackgroundEvents ?? DefaultConfiguration.trackBackgroundEvents,
+                customEndpoints: customEndpoints?.asCustomEndpoints(),
+                additionalConfig: [
+                    CrossPlatformAttributes.ddsource: "react-native",
+                    CrossPlatformAttributes.sdkVersion: SdkVersion
+                ],
+                configurationForTelemetry: configurationForTelemetry?.asConfigurationForTelemetry(),
+                nativeViewTracking: nativeViewTracking ?? DefaultConfiguration.nativeViewTracking,
+                nativeInteractionTracking: nativeInteractionTracking ?? DefaultConfiguration.nativeInteractionTracking,
+                verbosity: verbosity,
+                proxyConfig: proxyConfig?.asProxyConfig(),
+                serviceName: serviceName,
+                firstPartyHosts: firstPartyHosts?.asFirstPartyHosts() ?? DefaultConfiguration.firstPartyHosts
+            )
+        }
+        throw ProgrammerError(description: "JSON configuration file is missing top-level \"configuration\" key.")
+    }
+}
+
+extension NSString? {
+    func asTrackingConsent() -> TrackingConsent {
+        let trackingConsent: TrackingConsent
+        switch self?.lowercased {
+        case "pending":
+            trackingConsent = .pending
+        case "granted":
+            trackingConsent = .granted
+        case "not_granted":
+            trackingConsent = .notGranted
+        default:
+            trackingConsent = .pending
+        }
+        return trackingConsent
```

Comment:
/nit
```suggestion
        switch self?.lowercased {
        case "pending":
            return .pending
        case "granted":
           return .granted
        case "not_granted":
            return .notGranted
        default:
            return .pending
        }
```

Same for the other ones.